contracts/wrapper/NameWrapper.sol

```diff
@@ -527,7 +538,7 @@ contract NameWrapper is
         bytes memory name = _saveLabel(parentNode, node, label);
         expiry = _checkParentFusesAndExpiry(parentNode, node, fuses, expiry);
 
-        if (ownerOf(uint256(node)) == address(0)) {
```

Comment:
Maybe we could incorporate the `isWrapped` check into `onlyTokenOwner`? That way we avoid duplicate operations and get it everywhere we need.

Reply:
We already check ens.owner(node) in canCallSetSubnodeOwner, so I don't thin this is necessary?

Reply:
Imho, the check here (i.e., `isWrapped`) is slightly different from the one in `canCallSetSubnodeOwner`. Specifically

+ The [ens owner check](https://github.com/ensdomains/ens-contracts/blob/19582673577cc7b97b6db4223e0f523f04bda514/contracts/wrapper/NameWrapper.sol#L683) in `canCallSetSubnodeOwner` guarantees that a `parentNode` cannot create subnodes if its `CANNOT_CREATE_SUBDOMAIN` is burnt

+ The check here is to guarantee that, for a given `node`, every time it has a fuse to burn, the `ens.owner(node)` should be `address(this)`. It is to prevent some potential hacks.

I am not really an ENS professional, and hence may be wrong here. Please kindly correct me if so.